java/src/main/java/ai/rapids/cudf/CuFile.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.rapids.cudf;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+
+/**
+ * JNI wrapper for accessing the cuFile API.
+ */
```

Comment:
Could we add in some documentation about how this API is not always available right now, and that if might fail to load.

Also could we get an API to be able to ask it at runtime if it failed to load or not.  That way we could fall back to other processing paths if it is not able to run.

And finally could we have a warning about how the APIs here are experimental and might change.  I am a little concerned about performance tuning in the future.  The cuFile API has a lot of things about registering buffers, etc.  It sounds like if we are going to do a lot of small transfers then it might be good to do things differently like register a pool of buffers and reuse them.  I'm not sure that is what we want for the Spark use case, but I then again until we do performance testing I also don't know about that.

Reply:
Added wording for 1 and 3.

For 2, right now it'll throw an exception if unable to load. Are you suggesting not throwing the exception, and instead allow the caller to query the availability?

Reply:
I personally would prefer that.  In the regular CUDF java APIS we log a warning if we cannot load the libraries and then an exception is thrown by java if someone tries to use a native API when the native library is not loaded.  This allows for consistency, but would also let someone realize that cuFile is not available for what ever reason and fall back to a different option.